sympy/calculus/util.py

```diff
@@ -709,6 +709,133 @@ def is_convex(f, *syms, **kwargs):
         return False
     return True
 
+
+def stationary_points(f, symbol, domain = S.Reals):
+    """
+    Returns the stationary points of a function in the given domain.
+
+    Parameters
+    ==========
+
+    f : Expr
+        The concerned function.
+    symbol : Symbol
+        The variable for which the stationary points are to be determined.
+    domain : Interval
+        The domain over which the stationary points have to be checked.
+        If unspecified, S.Reals will be the default domain.
+
+    Examples
+    ========
+
+    >>> from sympy import Symbol, S, sin, log, pi, pprint
+    >>> from sympy.sets import Interval
+    >>> from sympy.calculus.util import stationary_points
+    >>> x = Symbol('x')
+
+    >>> stationary_points(1/x, x, S.Reals)
+    EmptySet()
+
+    >>> pprint(stationary_points(sin(x), x), use_unicode=False)
+              pi                              3*pi
+    {2*n*pi + -- | n in Integers} U {2*n*pi + ---- | n in Integers}
+              2                                2
+
+    >>> stationary_points(sin(x),x, Interval(0, 4*pi))
+    {pi/2, 3*pi/2, 5*pi/2, 7*pi/2}
+
+    """
+    from sympy import solveset, diff
+
+    if isinstance(domain, EmptySet):
+        return S.EmptySet
+
+    #domain = continuous_domain(f, symbol, domain)
+    set = solveset(diff(f, symbol), symbol, domain)
+
+    return set
+
+
+def maximize(f, symbol, domain = S.Reals):
```

Comment:
I was wondering if we can use `symbol = None` which might make things easier in case of multiple variables. We can extract all the symbols using `.atoms()` and then work with them.

Reply:
Yes, I think that makes sense, especially if there is only a single symbol in the expression. Does it work with multiple symbols? `.free_symbols` is also an option to extract them. In case a single one it should be straightforward.

Reply:
Btw, no space around the = for optional arguments.

Reply:
> Btw, no space around the = for optional arguments.

Yes, I'll change it.
>Does it work with multiple symbols?

As of know, it finds max with respect to given variable only. Problem may arise if the user is giving an expression with parameters other than the variables. For eg. `x**2 + y**2 + k` and wants to find maximum (`k` being a parameter).

Reply:
OK! I think it makes sense to be able to leave the symbol out if there is only one `free_symbol`, so check for that and if more than one raise an exception. Can be left for later as well though as it will not break the interface in any way.